pkg/services/rendering/auth.go

```diff
@@ -18,15 +24,52 @@ type RenderUser struct {
 	OrgRole string
 }
 
+type renderJWT struct {
+	RenderUser *RenderUser
+	jwt.RegisteredClaims
+}
+
 func (rs *RenderingService) GetRenderUser(ctx context.Context, key string) (*RenderUser, bool) {
-	val, err := rs.RemoteCacheService.Get(ctx, fmt.Sprintf(renderKeyPrefix, key))
+	var (
+		from   string
+		exists bool
+		err    error
+		start  = time.Now()
+	)
+
+	defer func() {
+		success := strconv.FormatBool(err == nil && exists)
+		metrics.MRenderingUserLookupSummary.WithLabelValues(success, from).Observe(float64(time.Since(start)))
+	}()
```

Comment:
I'm not very good at writing beautiful code that contains metric observations, so any kind of feedback about how to make this more readable/maintainable will be really appreciated and more than welcome.

Reply:
Maybe you can create two functions to retrieve the information through jwt or cache that return `(*RenderUser, bool)` and send directly the metrics with that result:

```go
start := time.Now()
if looksLikeJWT(key) && rs.features.IsEnabled(featuremgmt.FlagRenderingOverJWT) {
  user, success := getJWTUser(blah)
  metrics.MRenderingUserLookupSummary.WithLabelValues(success, "jwt").Observe(float64(time.Since(start)))
  return user, success
}

user, success := getCacheUser(blah)
metrics.MRenderingUserLookupSummary.WithLabelValues(success, "cache").Observe(float64(time.Since(start)))
return user, success
```

Reply:
Great idea, thanks @spinillos! 🙏🏻